x-pack/packages/observability/observability_utils/object/unflatten_object.test.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { unflattenObject } from './unflatten_object';
+
+describe('unflattenObject', () => {
+  it('unflattens deeply nested objects', () => {
+    expect(unflattenObject({ 'first.second.third': 'third' })).toEqual({
+      first: {
+        second: {
+          third: 'third',
+        },
+      },
+    });
+  });
+
+  it('does not unflatten arrays', () => {
+    expect(
+      unflattenObject({
+        simpleArray: ['0', '1', '2'],
+        complexArray: [{ one: 'one', two: 'two', three: 'three' }],
+        'nested.array': [0, 1, 2],
```

Comment:
btw, we should think about what this means in practice. e.g. let's say I have `child.id: [ 'a', 'b' ]`. I would expect the result to be `{ child: { id: [ 'a', b' ] }`. but this code results in `{ 'child.id': [ 'a, 'b' ] }` which I don't think is what we want.

Reply:
This line results in [`{ nested: { array: [0,1,2] } }`](https://github.com/elastic/kibana/pull/195242/files/8e702803beca5dee7991f8282daf2b74cac95e67#diff-65c942c594a513fb9b343f5f9d792274390effe5abbaff273689ade6ed29f474R31), If I understood correclty, it would work as would expect it to in the `child.id: ['a','b']` scenario.

Reply:
Oh right, so it only flattens leaf nodes. Sorry, read too quickly. Actually can you add a test here what happens with plain objects in an array? (This happens with eg the histogram field)

Reply:
Done!